### ISSUE TITLE
Proxy support

### ISSUE THREAD

**Issue by TheNetworkGuy** [TheNetworkGuy/netbox-zabbix-sync#8]
Currently there is no proxy support. I'm still debating on how to implement this. This could be a custom field, however i would rather see config context. The thought process behind this is that Zabbix proxies usually account for all devices in a site, so assigning a single proxy object to a site saves a lot of work instead of manually assigning it to each new device at a site / rack / etc.

**Comment by TheNetworkGuy** [comment]
I've updated the code to include proxy support. To simplify the transition i've made a new -p flag that is required when:
A. Netbox does not have a Zabbix proxy configured
B. Zabbix host object does van a proxy configured
C. You want to sync the devices up, meaning the proxy link will get deleted in Zabbix

This is the only case in which the -p flag will help. In all other use cases the -p flag can be omitted. For example:
 - When you want to add a proxy from within Netbox
 - When you want to change the proxy from within Netbox

TLDR: Deleting proxy links when a proxy is NOT specified within Netbox is locked behind the -p flag. This can be changed in the near future but is for now a design choice, not a bug.

**Comment by TheNetworkGuy** [comment]
Closing this one since its been open for almost half a year now.
